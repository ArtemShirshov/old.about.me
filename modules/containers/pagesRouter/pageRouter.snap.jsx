import React from 'react';
import renderer from 'react-test-renderer';

import {PagesRouter} from './pagesRouter';

jest.mock('@reach/router', () => ({
  Router: 'Router',
  Match: () => 'Match',
}));

jest.mock('containers/dynamicPage/dynamicPage', () => ({
  DynamicPageConnected: 'DynamicPageConnected',
}));
jest.mock('constants/routes/routes', () => ({
  StaticRoutes: [
    {
      path: '/',
      page: 'home',
      component: 'HomeConnected',
    },
  ],
}));

describe('<PagesRouter />', () => {
  const props = {
    setPage: () => {},
  };

  it('should render pagesRouter page ', () => {
    const tree = renderer.create(<PagesRouter {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
