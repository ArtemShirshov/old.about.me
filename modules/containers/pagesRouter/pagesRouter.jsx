// @flow
import React, {Fragment, PureComponent, type Node} from 'react';
import {connect} from 'react-redux';
import {Router, Match} from '@reach/router';

import {StaticRoutes} from 'constants/routes/routes';
import type {VoidFunctionType} from 'constants/flow/flowTypes';

import {Header} from 'components/header/header';
import {setPage, type PageType} from 'reducers/route/index';

type PropsType = {
    setPage: VoidFunctionType<PageType>,
};

type StateType = {
    hasError: boolean,
};

/**
 *  Class for PagesRouter
 */
export class PagesRouter extends PureComponent<PropsType, StateType> {
    state = {
        hasError: false,
    };

    /**
     * Catch JavaScript errors anywhere in their child component tree
     *
     * @param {{}} error - error
     * @param {{}} errorInfo - error info
     */
    componentDidCatch(error: {}, errorInfo: {}) {
        this.setState({hasError: true});
    }

    /**
     * Set match page and params
     *
     * @param {string} page - page name
     * @param {{}} match - params from url
     */
    setPage = (page: string, match: {}) => {
        this.props.setPage({page, params: match});
    };

    /**
     * Render jsx to html
     *
     * @returns {Node} Rendered react component
     */
    render(): Node {
        if (this.state.hasError) {
            return <p>Something went wrong.</p>;
        }

        return (
            <Fragment>
                <Header />
                <Router>
                    {StaticRoutes.map(route => (
                        <route.component path={route.path} key={route.path} page={route.page} />
                    ))}
                </Router>
                {StaticRoutes.map(route => (
                    <Match path={route.path} key={route.path}>
                        {props => Boolean(props.match) && this.setPage(route.page, props.match)}
                    </Match>
                ))}
            </Fragment>
        );
    }
}

export const mapDispatchToProps = {setPage};

export const PagesRouterConnected = connect(
    null,
    mapDispatchToProps,
)(PagesRouter);
