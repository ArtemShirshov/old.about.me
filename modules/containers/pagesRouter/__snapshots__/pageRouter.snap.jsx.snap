// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PagesRouter /> should render pagesRouter page  1`] = `
Array [
  <Router>
    <HomeConnected
      page="home"
      path="/"
    />
    <DynamicPageConnected
      default={true}
    />
  </Router>,
  "Match",
]
`;
